roles/openshift_inventory/tasks/main.yml

```diff
@@ -0,0 +1,22 @@
+---
+- no_log: True  # keep task descriptions legible
+  block:
+
+  - name: Standardize on latest variable names
+    set_fact:
+      deployment_type: "{{ openshift_deployment_type | default(deployment_type) | default(None) }}"
+      openshift_deployment_type: "{{ openshift_deployment_type | default(deployment_type) | default(None) }}"
+
+  - name: Normalize openshift_release
+    set_fact:
+      openshift_release: "{{ openshift_release | string | regex_replace('^v', '') }}"
+    when: openshift_release is defined
+
+- name: Ensure a valid deployment type has been given |
+  when: >
+    openshift_deployment_type is not defined
+    or openshift_deployment_type not in known_openshift_deployment_types
+  fail:
+    msg: |-
+      Please set openshift_deployment_type to one of:
+      {{ known_openshift_deployment_types | join(', ') }}
```

Comment:
Good reference post, I couldn't find a summary that succinct before

> /me adds to `DOCS` bookmark folder

From you reference I now understand that the `|` pipe followed by a `-` hyphen means 
in
>  **Literal style** - turns newlines within the string into literal newlines, and adds one at the end.

With the `-` *chomping indicator* the `strip` action happens:

> **strip**: remove the line feed, remove the trailing blank lines.
So that will remove the last new-line character at the end of the string (recall from above that the `|` literal appends a `\n` to the string) and trim off any trailing blank lines.

*Oh YAML, you so crazy some times*